### DISCUSSION TITLE
Using x265 encoding on a Mac

### DISCUSSION THREAD

**Discussion by svwhisper** [ehendrix23/tesla_dashcam#195] [Q&A] [answered]
When I try to encode using x265, ffmpeg throws a lot of errors.  I've tried varying parameters to see if there's a combination that works, but no joy yet.  I realise I'm not providing any useful info here, but as a first step I thought it worth asking if anyone using a Mac is successfully encoding x265 and, if so, can you share the arguments to tesla_dashcam?

**Comment by ehendrix23** [comment]
I've done it with no issues. What kind of MAC? Intel or Apple CPU?

**Reply by svwhisper** [reply]
Great to hear.  I'm running Monterey on an Intel mac.  The command I invoke is 

`python3 /usr/local/lib/python3.9/site-packages/tesla_dashcam/tesla_dashcam.py --skip_existing --monitor --monitor_trigger /Volumes/Teslacam/ARCHIVE_UPLOADED --encoding x265  --output /Volumes/TeslaCombinedVideos /Volumes/Teslacam`

It immediately returns "GPU acceleration is enabled" and then starts to scan and process.  Simply changing "x265" to "x264" and everything works as expected.  The error eventually returned when requesting x265 is:

`	Creating movie /Volumes/TeslaCombinedVideos/2022-05-04T09-09-32_2022-05-04T09-20-43.mp4, please be patient.
			Error trying to create movie /Volumes/TeslaCombinedVideos/2022-05-04T09-09-32_2022-05-04T09-20-43.mp4. RC: 1
			Command: ['ffmpeg', '-loglevel', 'info', '-f', 'concat', '-safe', '0', '-i', '/var/folders/nj/29t76m7n7hbgpnq6fskm9g600000gn/T/tmpky6pc8ib.txt', '-i', '/var/folders/nj/29t76m7n7hbgpnq6fskm9g600000gn/T/tmpjs1vf13r.txt', '-map_metadata', '1', '-map_chapters', '1', '-movflags', '+faststart', '-c', 'copy', '-metadata', 'creation_time=2022-05-03T23:09:32.000000Z', '-metadata', 'description=Created by tesla_dashcam v0.1.20', '-metadata', 'title=SENTRY: 05/04/22 09:20:41', '-metadata', 'location=-38.3397+143.5860', '-metadata', 'location-eng=-38.3397+143.5860', '-y', '/Volumes/TeslaCombinedVideos/2022-05-04T09-09-32_2022-05-04T09-20-43.mp4']
			Error: ffmpeg version 4.3.1 Copyright (c) 2000-2020 the FFmpeg developers
  built with Apple clang version 12.0.0 (clang-1200.0.32.28)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.3.1_9 --enable-shared --enable-pthreads --enable-version3 --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librtmp --enable-libspeex --enable-libsoxr --enable-videotoolbox --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fd32a010e00] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fd32a010e00] moov atom not found
[concat @ 0x7fd32a008200] Impossible to open 'file:/Volumes/TeslaCombinedVideos/2022-05-04T09-09-32.mp4'
/var/folders/nj/29t76m7n7hbgpnq6fskm9g600000gn/T/tmpky6pc8ib.txt: Invalid data found when processing input

`

**Comment by ehendrix23** [comment, marked as answer]
Error is stating it was impossible to open file /Volumes/TeslaCombinedVideos/2022-05-04T09-09-32.mp4. Based on the information, that is the location to put the created file.
Check if /Volumes/TeslaCombinedVideos is accessible for you. Maybe open a command line and run:
```
touch /Volumes/TeslaCombinedVideos/2022-05-04T09-09-32.mp4
```
To see if that works. Understand it works with x264 but just want to make sure as that is what it seems to be complaining about.

Also, see to get a newer version of ffmpeg: https://www.osxexperts.net/

**Reply by svwhisper** [reply]
"touch" works without error.  I'll try a new version of ffmpeg.  THANKS

**Comment by svwhisper** [comment]
Updated via "brew install ffmpeg".... seemingly hundreds of dependencies updated as well... now x265 encoding works a treat.

Thank you
